### PR TITLE
Segment group file format selection for serialization 

### PR DESCRIPTION
**Pull request by PaulHax**
VolView will save "segment group" images into a file format configurable by a config.JSON file.  `deserialize` will read whatever format it finds in the .zip file.

This is the config json to change the saved format
```
{
  "io": {
    "segmentGroupSaveFormat": "nrrd"
  }
}
```
To test: Can drag and drop this config file on VolView before saving. 
[config-save-format.json](https://github.com/Kitware/VolView/files/13621043/config-save-format.json)

Possible formats here:
https://github.com/InsightSoftwareConsortium/itk-wasm/blob/main/packages/image-io/typescript/src/extension-to-image-io.ts

TODO

- [x] Direction consines not right after round trip for prostate
- [x] Stop itk-wasm:writeImage from detaching image array when passing to worker
- [x] Bundle @itk-wasm/image-io wasms in build
- [x] Use @itk-wasm/image-io for loading base images, not just segment groups
~- [ ] Add GUI to set format, maybe~  Follow up PR to add GUI.

### CODE REVIEW

**Review by jadh4v — COMMENTED** [review]
I am wondering if there's going to be a conflict in the future regarding the segmentation information that is stored in the session's json file vs the embedded info stored inside .nrrd and other fancier file formats that we might support in the future.

File: package.json [hunk 1]
```diff
@@ -126,4 +126,4 @@
       "eslint"
     ]
   }
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (PaulHax) [reply package.json:129]: If prettier gets at it, it adds a newline.

File: src/store/segmentGroups.ts [hunk 2]
```diff
@@ -348,7 +374,7 @@ export const useSegmentGroupStore = defineStore('segmentGroup', () => {
         const metadata = metadataByID[id];
         return {
           id,
-          path: `labels/${id}.vti`,
+          path: `labels/${id}.${saveFormat.value}`,
```
[the remark this answers is not in the document]
Comment (PaulHax) [reply src/store/segmentGroups.ts:377]: I like both ideas.  Might rather move the format selection from Settings to the segment group UI cluster under annotation tab.

**Comment by PaulHax** [comment]
This is good to go by me.  Followup PR for format selection and docs.

File: src/store/segmentGroups.ts [hunk 3]
```diff
@@ -27,6 +33,36 @@ import {
...
+  // Transpose the direction matrix to fix bug in @itk-wasm/image-io.writeImage
+  // Remove when @itk-wasm/image-io version is above 0.5.0 https://github.com/InsightSoftwareConsortium/itk-wasm/commit/ad9ca85eedc47c9d3444cf36859569c529886bde
+  const oldDirection = [...itkImage.direction];
+  const { dimension } = itkImage.imageType;
+  for (let idx = 0; idx < dimension; ++idx) {
+    for (let idy = 0; idy < dimension; ++idy) {
+      itkImage.direction[idx + idy * dimension] =
+        oldDirection[idy + idx * dimension];
+    }
+  }
```
Comment (PaulHax) [inline src/store/segmentGroups.ts:60]: Need to remember to nuke this workaround after bumping @itk-wasm/image-io

Comment (floryst) [reply src/store/segmentGroups.ts:60]: Can you add a TODO prefix to the comment (or submit an issue after this PR is merged)? That way it's easier to find TODOs that need addressing in the codebase.

File: src/store/segmentGroups.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (floryst) [reply src/store/segmentGroups.ts:389]: It seems more relevant for the save format to be in the save dialog.

[the remark this answers is not in the document]
Comment (PaulHax) [reply src/store/segmentGroups.ts:389]: Yes.  Lets disscuss in an issue for a folow up PR

File: src/store/segmentGroups.ts (hunk 3 above)
Comment (PaulHax) [reply src/store/segmentGroups.ts:60]: I'll submit a issue after this PR.

File: src/store/segmentGroups.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jadh4v) [reply src/store/segmentGroups.ts:390]: Perhaps you'd want to create an issue if you're addressing it in a separate PR before resolving this thread.

[the remark this answers is not in the document]
Comment (jadh4v) [reply src/store/segmentGroups.ts:390]: Ok, created https://github.com/Kitware/VolView/issues/536 to track this.

File: src/io/readers.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import vtkITKImageReader from '@kitware/vtk.js/IO/Misc/ITKImageReader';
+import { convertItkToVtkImage } from '@kitware/vtk.js/Common/DataModel/ITKHelper';
 import { readImageArrayBuffer, extensionToImageIO } from 'itk-wasm';
+import { readImage } from '@itk-wasm/image-io';
 import { FileReaderMap } from '.';
 
-import { readFileAsArrayBuffer } from './io';
 import { stlReader, vtiReader, vtpReader } from './vtk/async';
```
Comment (jadh4v) [inline src/io/readers.ts:7]: Looks like `itk-wasm/mesh-io` also has a similar generic reader for polydata? Maybe in the future we could get rid of these special readers as well.

File: src/store/dicom-web/dicom-web-store.ts [hunk 5]
```diff
@@ -203,6 +203,7 @@ export const useDicomWebStore = defineStore('dicom-web', () => {
         state: 'Done',
       };
     } catch (error) {
+      console.error(error);
```
Comment (jadh4v) [inline src/store/dicom-web/dicom-web-store.ts:206]: Do you want to keep this, or was just added during debugging?

**Review by jadh4v — APPROVED** [review]
LGTM, apart from a few general comments.

File: src/io/readers.ts (hunk 4 above)
Comment (PaulHax) [reply src/io/readers.ts:7]: Yes, cool.  VolView's mesh support is just vestigial at the moment.

File: src/store/dicom-web/dicom-web-store.ts (hunk 5 above)
Comment (PaulHax) [reply src/store/dicom-web/dicom-web-store.ts:206]: Good check, did want to keep actualy.

**Review by floryst — APPROVED** [review]
LGTM. Haven't tested functionality yet.

**Pull request merged**